src/test/java/org/elasticsearch/index/analysis/SkroutzGreekStemmerTest.java

```diff
@@ -56,23 +56,25 @@ public void testSkroutzGreekStemmerExceptionsZero()
     Field field = stemmer.getClass().getDeclaredField("stepZeroExceptions");
     field.setAccessible(true);
 
-    Map<String, char[]> stepZeroExceptions = (Map) field.get(stemmer);
-
-   for (Map.Entry<String, char[]> entry : stepZeroExceptions.entrySet()) {
-     Assert.assertTrue(entry.getKey().length() >= entry.getValue().length,
-             "Make sure that the stem is shorter/equal than the original.");
-   }
+    CharArrayMap stepZeroExceptions = (CharArrayMap) field.get(stemmer);
+    CharArrayMap.EntryIterator it = stepZeroExceptions.entrySet().iterator();
+    while(it.hasNext()) {
```

Comment:
@gmetal true, I just think that calling it `nextKey` is a bit confusing :) I'm also more used in methods such as [each](https://ruby-doc.org/core-2.2.0/Array.html#method-i-each) that take care of the gritty details!